pkg/template/template.go

```diff
@@ -0,0 +1,232 @@
+package template
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/cli/go-gh/pkg/tableprinter"
+	"github.com/mattn/go-runewidth"
+	"github.com/mgutz/ansi"
+)
+
+type Template struct {
+	colorEnabled bool
+	output       io.Writer
+	tmpl         *template.Template
+	tp           tableprinter.TablePrinter
+	width        int
+}
+
+func New(w io.Writer, width int, colorEnabled bool) Template {
+	return Template{
+		colorEnabled: colorEnabled,
+		output:       w,
+		tp:           tableprinter.New(w, true, width),
+		width:        width,
+	}
+}
+
+func (t *Template) Parse(tpl string) error {
```

Comment:
This is the only change from the `cli` version of this package. `Parse` now needs to be explicitly called rather than implicitly being called from within `Execute`. I made this change to reduce the number of arguments to `New` and `Execute`. Additionally, I find the explicitness of it to be easier to follow.